process/testing/winfv-cni-plugin/aso/vmss.sh

```diff
@@ -112,6 +113,51 @@ EOF
 chmod +x ./scp-from-windows.sh
 
   pause-for-debug
+  echo "show_connections done."; echo
+}
+
+function retry-ssh() {
+  local SSH_CMD=$1
+  local RETRY_INTERVAL=30         # Seconds between retries
+  local MAX_DURATION=300
+
+  # Tracking time
+  START_TIME=$(date +%s)
+
+  while true; do
+    echo "Attempting $SSH_CMD..."
+    if $SSH_CMD; then
+        echo "SSH command succeeded."
+        return 0
+    else
+        echo "SSH command failed. Running show_connections and retrying in $RETRY_INTERVAL seconds..."
+        show_connections  # Replace with your actual command or function
+    fi
+
+    sleep $RETRY_INTERVAL
+
+    CURRENT_TIME=$(date +%s)
+    ELAPSED_TIME=$((CURRENT_TIME - START_TIME))
+
+    if [ $ELAPSED_TIME -ge $MAX_DURATION ]; then
+        echo "Timeout reached after $((MAX_DURATION / 60)) minutes. Giving up."
+        exit 1
+    fi
+  done
+}
+
+# Azure may assign a different public IP to a VM even after it is marked as ready.
+# This function attempts to SSH into the VM multiple times to ensure it is accessible.
+function confirm-nodes-ssh() {
+  echo;echo "confirm nodes can be accessed by ssh..."
+  show_connections
+  retry-ssh "${MASTER_CONNECT_COMMAND} echo"
+  retry-ssh "${WINDOWS_CONNECT_COMMAND} -Command 'echo'"
+  echo "sleep 30 seconds..."
+  sleep 30
+  retry-ssh "${MASTER_CONNECT_COMMAND} echo"
+  retry-ssh "${WINDOWS_CONNECT_COMMAND} -Command 'echo'"
```

Comment:
We are essentially doubling the retry-ssh() timeout here, right? Any specific reason to call it twice instead of using a 600 timeout and calling it just once (per node)?

Reply:
Oh I see, so the _CONNECT_COMMANDs may change between the calls? Gotcha, makes sense... Could you please add a comment stating that?